api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -233,8 +235,12 @@ def serialize_estimated_shipping_rates(shipping_rates)
             ).serializable_hash
           end
 
+          def metadata_params
+            params.permit(public_metadata: {}, private_metadata: {})
+          end
+
           def add_item_params
-            params.permit(:quantity, :variant_id, public_metadata: {}, private_metadata: {}, options: {})
```

Comment:
@aleksandarpetrushev why do we need this change? let's keep everything in `add_item_params` if possible

Reply:
@damianlegawiec Sure, I will revert. Made that change because we use *_metadata params in 2 endpoints and this way we would need to call add_item_params in #create